### ISSUE TITLE
Implement non-function type alias support in the VM/AOT

### ISSUE THREAD

**Issue by leafpetersen** [dart-lang/sdk#44954]
This issue tracks support of the [non-function type alias feature](https://github.com/dart-lang/language/blob/master/accepted/future-releases/generalized-typedef-2018/feature-specification.md).  See the [enclosing meta-issue](https://github.com/dart-lang/sdk/issues/44951) for details.

I don't expect much work here, but we should validate that debugging/error messages work as desired, and that the backend doesn't choke.

cc @a-siva @mkustermann @mraleph  @eernstg @mit-mit @franklinyow

**Comment by a-siva** [comment]
As indicated by you there is no work in the VM for implementing this feature as the front end will account for substituting the type alias.
With regards to debugging/error messages working as desired, what is the expectation? I believe currently for all typedefs the VM only shows the substituted type, this is probably true for DDC too.

**Comment by leafpetersen** [comment]
I have no particular expectations, this issue is just to provide a sign off point for us to validate that the backend doesn't choke, and that any desired functionality is implemented.

I think that showing the substituted type in error messages is fine. 

I don't know how much of these are directly VM related, but things that I can imagine being worth validating:
  - Hot reload works in a reasonable manner if a typedef is edited, or a use of a typedef is edited
  - Debugging works in a reasonable manner (i.e. doesn't crash)
  - Global type optimizations aren't confused by uses of typedefs (if everything gets substituted away early, that should be trivially true)

Basically, whenever you're comfortable with the native backend support for this feature, feel free to close this out.

**Comment by leafpetersen** [comment]
@a-siva should I close this out or is there any work we have planned here?

**Comment by a-siva** [comment]
I had it open just to ensure that the tests passed once CFE completed the implementation.

**Comment by a-siva** [comment]
Closing this as no work is necessary in the VM, we will track all individual test failures by opening fresh issues for each failure.
